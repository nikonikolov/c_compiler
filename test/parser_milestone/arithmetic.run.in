# 2 "arithmetic.run.in" 3 4
int x()
{}


int g, b, d, f=a;


int zz(int a, int b, int c)
{
  int a, b=a+b, c=(a-A)*(a+a) ;

  if(a==b)
    a=b;
  else if(a==c) ;
  else if(a==d) {

  }
  else{
    int fsdfsdfs;
    return a==b;
  }

	while(a==b){
		a=b;
	}

	for(i=a; i<a; i=b){
		a=b;
	}  
	return a==b;
}