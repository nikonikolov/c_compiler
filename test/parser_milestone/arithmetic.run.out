FUNCTION : x
SCOPE
VARIABLE : g
VARIABLE : b
VARIABLE : d
VARIABLE : f
FUNCTION : zz
    PARAM : a
    PARAM : b
    PARAM : c
SCOPE
    VARIABLE : a
    VARIABLE : b
    VARIABLE : c
    SCOPE
        VARIABLE : fsdfsdfs
    SCOPE
    SCOPE
